src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/IProvenBlockHeaderStore.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Database of <see cref="ProvenBlockHeader"/>s.
+    /// </summary>
+    public interface IProvenBlockHeaderStore : IDisposable
```

Comment:
https://github.com/stratisproject/StratisBitcoinFullNode/issues/2142

```
Apro:
includes initialization and rewinding
```

Please add rewinding

Reply:
Here's a note to handle the rewind in the cache layer - https://github.com/stratisproject/StratisBitcoinFullNode/issues/2142#issuecomment-422341179.

Reply:
I'm not sure if rewind can be only in cache layer. Most likely it should be in both.

Reply:
If it's in both, I think it's best to get this PR pushed and when I start on the caching layer I will include rewind in both the store and repository classes.

Reply:
I agree that caching layer should be done after DB layer is completed. 
But the db layer rewind is a part of this PR.